core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -597,6 +598,20 @@ private[spark] class AppStatusListener(
           None
       }
       task.errorMessage = errorMessage
+      task.failureReason = event.reason match {
```

Comment:
As I mentioned in another place, `message` in FailureReason can be redundant.
If we can remove it, we can just parse error message and build `FailureReason` in `computeFailureSummary` regardless of `event.reason`.

Reply:
Yes, in previous version, I directly passed error message to build failure reason, but I found it hard, complex and error-prone. This due to different failure reason has different format:
1. `ExceptionFailure` is most common and easy one. The message is only exception's message, we can parse it from errorString.
2. `ExecutorLostFailure` errorString has format like `ExecutorLostFailure (executor 72 exited caused by one of the running tasks) Reason: Executor heartbeat timed out after 123153 ms`. We can split the message out using keyword `Reason:`
3. `FetchFailed` errorString has format like `FetchFailed($bmAddressString, shuffleId=$shuffleId, mapId=$mapId, reduceId=$reduceId, message=\n$message\n)`. We need to parse message from this.

In summary, parsing will rely on and couple with specific format of different failure reason. If in the future, the format changed, it's easy to break. So I chose the solution to directly get from TaskEndReason.

Reply:
@sarutak Any more comments on this?